### PR TITLE
Staking Documentation Improvements; Sidebar Cleanup

### PR DESCRIPTION
**Pull request by KPrasch**

### CODE REVIEW

File: docs/source/guides/staking_guide.rst [hunk 1]
```diff
@@ -47,6 +47,17 @@ All staking-related operations are performed through the ``nucypher stake`` comm
...
+Staking Overview
+-----------------
+
+Most stakers will use the following commands in order:
+
+1. Create a new persistent stakeholder configuration
+2. Initialize a new stake
+3. List all Stakes
+4. Bond a worker to the stake
+5. Divide an existing Stake
```
Comment (cygnusv) [inline docs/source/guides/staking_guide.rst]: Is divide something that stakers will have to do? I still don't know what are good use cases for dividing stake. @michwill @szotov ?

File: docs/source/guides/staking_guide.rst [hunk 2]
```diff
@@ -47,6 +47,17 @@ All staking-related operations are performed through the ``nucypher stake`` comm
...
 
+Staking Overview
+-----------------
+
+Most stakers will use the following commands in order:
+
+1. Create a new persistent stakeholder configuration
+2. Initialize a new stake
+3. List all Stakes
+4. Bond a worker to the stake
```
Comment (cygnusv) [inline docs/source/guides/staking_guide.rst]:
```suggestion
4. Bond a worker to the staker
```

File: docs/source/guides/staking_guide.rst [hunk 3]
```diff
@@ -127,6 +138,24 @@ Interactive Method
     | 2 | 0xbb03 | 0xbb04 | 0 | 30000 NU | 39 periods . | Aug 09 12:15:16 CEST - Sep 16 12:15:16 CEST
 
 
+*Bond an Ursula to a Stake*
```
Comment (cygnusv) [inline docs/source/guides/staking_guide.rst]:
```suggestion
*Bond an Ursula to a Staker*
```

File: docs/source/guides/staking_guide.rst (hunk 1 above)
Comment (mswilkison) [reply docs/source/guides/staking_guide.rst]: Maybe you're close to the max_stake and want to restake your rewards?

File: nucypher/config/keyring.py [hunk 4]
```diff
@@ -663,7 +663,7 @@ def validate_password(password: str) -> List:
         """
         rules = (
             (bool(password), 'Password must not be blank.'),
-            (len(password) >= 16, 'Password is too short, must be >= 16 chars.'),
+            (len(password) >= 3, 'Password is too short.'),  # TODO: Increase this value for mainnet.
```
Comment (mswilkison) [inline nucypher/config/keyring.py]:
```suggestion
            (len(password) >= 16, 'Password is too short, must be >= 16 chars.'),
```

File: docs/source/guides/staking_guide.rst (hunk 1 above)
Comment (szotov) [reply docs/source/guides/staking_guide.rst]: when you want to extend/prolong only part of your stake

File: docs/source/guides/staking_guide.rst (hunk 3 above)
Comment (cygnusv) [reply docs/source/guides/staking_guide.rst]: This isn't resolved yet!

**Pull request merged**